src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1235,6 +1245,17 @@ public SolutionState AddDocuments(ImmutableArray<DocumentInfo> documentInfos)
 
                 var (newProjectState, compilationTranslationAction) = addDocumentsToProjectState(oldProject, newDocumentStatesForProject);
 
+                // PROTOTYPE: we add the edit to the projects driver, if we have one
+                if (_projectIdToGeneratorDriverMap.ContainsKey(oldProject.Id))
+                {
+                    GeneratorDriver driver = _projectIdToGeneratorDriverMap[oldProject.Id]!;
+
+                    var edits = newDocumentStatesForProject.SelectAsArray<T, PendingEdit>(s => new AdditionalFileAddedEdit(new AdditionalTextWithState(s)));
+                    driver = driver.WithPendingEdits(edits);
+
+                    _projectIdToGeneratorDriverMap = _projectIdToGeneratorDriverMap.SetItem(oldProject.Id, driver);
```

Comment:
I would have initially assumed this new map with the new driver must be passed to the new snapshot, not mutating this existing snapshot. The idea behind this is you can use the new driver and then ask for an updated generated compilation that's post-edit-applied, right?

Reply:
Ah, yes that was the intention. It gets copied by the ForkProject, but I hadn't considered that it'll mutate the existing snapshot to do so. I'll make a note to fix that.

---
In reply to: [388643118](https://github.com/dotnet/roslyn/pull/41789#discussion_r388643118) [](ancestors = 388643118)